### ISSUE TITLE
When I did the validation, preds was [1, 0, 0, 1].

### ISSUE THREAD

**Issue by FriedaSmith** [shamanez/BERT-like-is-All-You-Need#17]
When I did the validation, `preds` was **[1, 0, 0, 1]**.
**The validation command** is as follows

`python validate.py --data valid_test/iemocap/datasets --path models/valid_test/gru_sf/iemocap/checkpoints_0414/checkpoint_best.pt --task emotion_prediction --valid-subset test --batch-size 1 --eval-metric`

I studied the code in **airseq/criterions/emotion_prediction_cri.py**.  I found that the shape of **logits** is **torch.Size([4, 2])**, as shown below.

```
tensor([[-0.2599,  1.4415],
        [ 0.9081, -0.5650],
        [ -1.2503, 1.2987],
        [ 1.0385, -1.1427]])
```
I want to know what **ncorrect** means. Why can `100.0 * ncorrect / nsentences` calculate accuracy?

```
        elif self.binary_target_iemocap:
            targets = targets.long()

            targets = torch.nn.functional.one_hot(targets, 4)

            logits = logits.view(-1, 2)
            targets = targets.view(-1)

            loss = F.nll_loss(
                F.log_softmax(logits, dim=-1, dtype=torch.float32),
                targets,
                reduction='sum',
            )

            preds = logits.max(dim=1)[1]

            ncorrect = (preds == targets).sum().item()
```

```
    def reduce_metrics(logging_outputs) -> None:

        """Aggregate logging outputs from data parallel training."""
        loss_sum = sum(log.get('loss', 0) for log in logging_outputs)
        ntokens = sum(log.get('ntokens', 0) for log in logging_outputs)
        nsentences = sum(log.get('nsentences', 0) for log in logging_outputs)
        sample_size = sum(log.get('sample_size', 0) for log in logging_outputs)

        metrics.log_scalar('loss', loss_sum / sample_size / math.log(2), sample_size, round=3)
        if sample_size != ntokens:
            metrics.log_scalar('nll_loss', loss_sum / ntokens / math.log(2), ntokens, round=3)

        if len(logging_outputs) > 0 and 'ncorrect' in logging_outputs[0]:
            ncorrect = sum(log.get('ncorrect', 0) for log in logging_outputs)
            metrics.log_scalar('accuracy', 100.0 * ncorrect / , nsentences, round=3)
```

**Comment by shamanez** [comment]
It is because for IEMOCAP I used the binary accuracy.  Binary accuracy is you try to predict whether that emotion is there or not. So if you have 4 emotions, you will get 8 output nodes. That is why I reshape it with (4,2).  ncorrect is a number of correctly predicted instances. 

When valid batch size is one, preds is a list with [1*4].

@FriedaSmith  hope this helps!

**Comment by FriedaSmith** [comment]
Thanks.
